### PR TITLE
Remove trait queries from Actor ctor.

### PR DESCRIPTION
**Pull request by pchote**
Inspired by @reaperrr's recent push to reduce trait queries: we are already looping over all the actor's traits in the ctor, so there is no need to query them again via the world. We can also remove the FrameEndTask for `staticTargetablePositions` because both conditions and trait creation notifications are now encapsulated within `Actor`, so we can directly evaluate it at the right time.

### CODE REVIEW

File: OpenRA.Game/Actor.cs [hunk 1]
```diff
@@ -140,43 +141,57 @@ internal Actor(World world, string name, TypeDictionary initDict)
...
+					{ if (trait is IRenderModifier t) renderModifiersList.Add(t); }
+					{ if (trait is IRender t) rendersList.Add(t); }
+					{ if (trait is IMouseBounds t) mouseBoundsList.Add(t); }
+					{ if (trait is IVisibilityModifier t) visibilityModifiersList.Add(t); }
+					{ if (trait is IDefaultVisibility t) defaultVisibility = t; }
+					{ if (trait is INotifyBecomingIdle t) becomingIdlesList.Add(t); }
+					{ if (trait is INotifyIdle t) tickIdlesList.Add(t); }
+					{ if (trait is ITargetable t) targetablesList.Add(t); }
+					{ if (trait is ITargetablePositions t) targetablePositionsList.Add(t); }
+					{ if (trait is ISync t) synchHashesList.Add(new SyncHash(t)); }
```
Comment (teinarss) [inline OpenRA.Game/Actor.cs]: Possible cleaner with a switch.
```
 switch (trait)
    {
        case IPositionable t:
           positionable = t;
           break;
        case IOccupySpace t:
           OccupiesSpace = t; 
           break;
.
.
.
```

Comment (pchote) [reply OpenRA.Game/Actor.cs]: That was my first idea too; it doesn't work because many traits implement more than one interface!

**Comment by pchote** [comment]
Rebased.

File: OpenRA.Game/Actor.cs [hunk 2]
```diff
@@ -140,44 +141,61 @@ internal Actor(World world, string name, TypeDictionary initDict)
...
+				var resolveOrdersList = new List<IResolveOrder>();
+				var renderModifiersList = new List<IRenderModifier>();
+				var rendersList = new List<IRender>();
+				var mouseBoundsList = new List<IMouseBounds>();
+				var visibilityModifiersList = new List<IVisibilityModifier>();
+				var becomingIdlesList = new List<INotifyBecomingIdle>();
+				var tickIdlesList = new List<INotifyIdle>();
+				var targetablesList = new List<ITargetable>();
+				var targetablePositionsList = new List<ITargetablePositions>();
+				var synchHashesList = new List<SyncHash>();
```
Comment (RoosterDragon) [inline OpenRA.Game/Actor.cs]: Nit: sync**h**HashesList

**Review by RoosterDragon — DISMISSED** [review]

File: OpenRA.Game/Actor.cs (hunk 2 above)
Comment (pchote) [reply OpenRA.Game/Actor.cs]: Whoops, fixed.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/dd8154656a89e56139e0cd63f4d6ee6eeae9f6f8...7874eec37eba220669f1cdaaff0af97258f04091

**Pull request merged**